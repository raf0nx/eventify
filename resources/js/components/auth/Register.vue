<template>
	<v-col
		cols="12"
		md="5"
		class="d-flex flex-column justify-center px-12"
		dark
	>
		<v-card elevation="0">
			<h1
				class="main-title font-weight-bold deep-purple--text text--darken-4"
			>
				Create a new account
			</h1>
			<v-form class="py-12" @submit.prevent="registerUser">
				<ValidationProvider
					v-slot="{ errors }"
					rules="required|alpha_spaces"
					name="Name"
				>
					<v-text-field
						color="deep-purple"
						type="text"
						label="Your Name"
						:error-messages="errors"
						v-model="formData.name"
					></v-text-field>
				</ValidationProvider>
				<ValidationProvider
					v-slot="{ errors }"
					rules="required|email"
					name="Email"
				>
					<v-text-field
						color="deep-purple"
						type="email"
						label="Your Email"
						:error-messages="errors"
						v-model="formData.email"
					></v-text-field>
				</ValidationProvider>
				<ValidationObserver>
					<ValidationProvider
						v-slot="{ errors }"
						rules="required|min:8"
						name="Password"
					>
						<v-text-field
							color="deep-purple"
							label="Password"
							type="password"
							:error-messages="errors"
							v-model="formData.password"
						></v-text-field>
					</ValidationProvider>
					<ValidationProvider
						v-slot="{ errors }"
						rules="required|confirmed:Password"
						name="Password Confirmation"
					>
						<v-text-field
							color="deep-purple"
							type="password"
							label="Confirm password"
							:error-messages="errors"
							v-model="formData.password_confirmation"
						></v-text-field>
					</ValidationProvider>
				</ValidationObserver>
				<v-btn
					color="purple"
					type="submit"
					class="white--text font-weight-bold mt-12"
					large
					block
					>Create account</v-btn
				>
			</v-form>
			<p class="text-center">
				Already have an account?
				<v-btn :to="{ name: 'Login' }" plain text small>
					Sign In
				</v-btn>
			</p>
		</v-card>
	</v-col>
</template>

<script>
	import { ValidationProvider, ValidationObserver } from "vee-validate";

	import AuthService from "../../services/AuthService";

	export default {
		components: {
			ValidationProvider,
			ValidationObserver,
		},
		data() {
			return {
				formData: {
					name: "",
					email: "",
					password: "",
					password_confirmation: "",
				},
			};
		},
		methods: {
			async registerUser() {
				this.$store.commit("utils/SET_LOADING", true);
				try {
					await AuthService.registerUser(this.formData);
					this.$router.push({ name: "Home" });
				} catch (error) {
					console.log(error);
				}
				this.$store.commit("utils/SET_LOADING", false);
			},
		},
	};
</script>

<style>
</style>