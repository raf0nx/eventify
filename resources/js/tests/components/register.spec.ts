import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";

import Register from "@components/auth/Register.vue";

const SIGN_UP = "Create a new account";

describe("Register.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<Register>;
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = shallowMount(Register, {
            localVue,
            vuetify
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should contain Sign Up header", () => {
        const header = wrapper.find("h1");
        expect(header.text()).toEqual(SIGN_UP);
    });

    it("Should trigger register method", async () => {
        const registerSpy = jest
            // @ts-ignore
            .spyOn(wrapper.vm, "registerUser")
            .mockResolvedValue(true);
        // @ts-ignore
        const isRegistered = await wrapper.vm.registerUser();

        expect(registerSpy).toBeCalled();
        expect(isRegistered).toBe(true);
    });
});
