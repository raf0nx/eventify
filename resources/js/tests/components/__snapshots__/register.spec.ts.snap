// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue Should match snapshot 1`] = `
"<v-col-stub cols=\\"12\\" md=\\"5\\" tag=\\"div\\" dark=\\"\\" class=\\"d-flex flex-column justify-center px-12\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" elevation=\\"0\\">
    <h1 class=\\"main-title font-weight-bold deep-purple--text text--darken-4\\">
      Create a new account
    </h1>
    <validationobserver-stub tag=\\"span\\" vid=\\"obs_0\\"></validationobserver-stub>
    <p class=\\"text-center\\">
      Already have an account?
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" to=\\"[object Object]\\" small=\\"true\\" plain=\\"true\\" text=\\"true\\" type=\\"button\\">
        Sign In
      </v-btn-stub>
    </p>
  </v-card-stub>
</v-col-stub>"
`;
